<?xml version="1.0" ?>
<sdf version="1.5">
  <world name="default">
    <scene>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.700000 0.700000 0.700000 1.000000</background>
      <shadows>1</shadows>
    </scene>
    <physics type="ode">
      <gravity>0.000000 0.000000 -9.81</gravity>
      <magnetic_field>6e-06 2.3e-05 -4.2e-05</magnetic_field>
      <atmosphere type='adiabatic'/>
      <ode>
        <solver>
          <type>quick</type>
          <iters>100</iters>
          <precon_iters>0</precon_iters>
          <sor>1.300000</sor>
        </solver>
        <constraints>
          <cfm>0.000000</cfm>
          <erp>0.200000</erp>
          <contact_max_correcting_vel>50.000000</contact_max_correcting_vel>
          <contact_surface_layer>0.001000</contact_surface_layer>
        </constraints>
      </ode>
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>

    <!-- Light Source -->
    <include>
      <uri>model://sun</uri>
    </include>

    <model name="gplane">
      <pose>0.000000 0.000000 0.000000 0.000000 -0.000000 0.000000</pose>
      <link name="plane">
        <pose>0.000000 0.000000 0.000000 0.000000 -0.000000 0.000000</pose>
        <collision name="plane">
          <pose>0.000000 0.000000 0.000000 0.000000 -0.000000 0.000000</pose>
          <geometry>
            <plane>
              <normal>0.000000 0.000000 1.000000</normal>
            </plane>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>50.000000</mu>
                <mu2>50.000000</mu2>
                <fdir1>0.000000 0.000000 0.000000</fdir1>
                <slip1>0.000000</slip1>
                <slip2>0.000000</slip2>
              </ode>
            </friction>
            <bounce>
              <restitution_coefficient>0.000000</restitution_coefficient>
              <threshold>100000.000000</threshold>
            </bounce>
            <contact>
              <ode>
                <soft_cfm>0.000000</soft_cfm>
                <soft_erp>0.200000</soft_erp>
                <kp>1000000000.000000</kp>
                <kd>1.000000</kd>
                <max_vel>100.000000</max_vel>
                <min_depth>0.001000</min_depth>
              </ode>
            </contact>
          </surface>
          <laser_retro>0.000000</laser_retro>
        </collision>
        <gravity>1</gravity>
        <self_collide>0</self_collide>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
    </model>


    <include>
      <name>table1</name>
      <uri>model://Table</uri>
      <pose>0.750000 0.000000 0.000000 0.000000 -0.000000 0.000000</pose>
    </include>

    <include>
      <name>table2</name>
      <uri>model://Table</uri>
      <pose>-0.80 0.000000 0.000000 0.000000 -0.000000 0.000000</pose>
    </include>


    <!-- <include>
      <name>Vans_Cereal_Honey_Nut_Crunch_11_oz_box</name>
      <uri>model://Vans_Cereal_Honey_Nut_Crunch_11_oz_box</uri>
      <pose>0.75 0.000000 1.00000 0.000000 0.000000 1.5708 </pose>
    </include> -->

    <model name="box_01_model">
      <allow_auto_disable>0</allow_auto_disable>
      <pose>0.95 0.000000 1.16600 0.000000 0.000000 0</pose>
      <link name="box_01_body">
        <pose>0.000000 0.000000 0.000000 0.000000 0.000000 0.000000</pose>
        <inertial>
          <pose>0.000000 0.000000 -0.15000 0.000000 0.000000 0.000000</pose>          
          <inertia>
            <ixx>0.100000</ixx>
            <ixy>0.000000</ixy>
            <ixz>0.000000</ixz>
            <iyy>0.100000</iyy>
            <iyz>0.000000</iyz>
            <izz>0.100000</izz>
          </inertia>
          <mass>1.000000</mass>          
        </inertial>
        <collision name="box_01_geom">
          <pose>0.000000 0.000000 0.0 0.000000 -0.000000 0.000000</pose>
          <geometry>
            <box>
              <size>0.1 0.2 0.3</size>
            </box>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>0.5</mu>
                <mu2>0.5</mu2>
                <fdir1>0.000000 0.000000 0.000000</fdir1>
                <slip1>0.000000</slip1>
                <slip2>0.000000</slip2>
              </ode>
            </friction>
          
            <contact>
              <ode>
                <soft_cfm>0.000000</soft_cfm>
                <soft_erp>0.200000</soft_erp>
                <kp>100000.000000</kp>
                <kd>1.000000</kd>
                <max_vel>100.000000</max_vel>
                <min_depth>0.001000</min_depth>
              </ode>
            </contact>
          </surface>
          <laser_retro>0.000000</laser_retro>
        </collision>
        <visual name="box_01_geom_visual">
          <pose>0.000000 0.000000 0.000000 0.000000 -0.000000 0.000000</pose>
          <geometry>
            <box>
              <size>0.1 0.2 0.3</size>
            </box>
          </geometry>
          <material>
            <ambient> 0.2 0.2 0.2 1.0 </ambient>
          </material>
          <cast_shadows>1</cast_shadows>
          <laser_retro>0.000000</laser_retro>
          <transparency>0.000000</transparency>
        </visual>
        <gravity>1</gravity>
        <self_collide>0</self_collide>
        <kinematic>0</kinematic>
      </link>
      <static>0</static>
    </model>


    <include>
      <name>one_disposal_bin</name>
      <uri>model://disposal_bin</uri>
      <pose>0 0.9 0 0 0 0</pose>
    </include>

    <include>
      <name>two_disposal_bin</name>
      <uri>model://disposal_bin</uri>
      <pose>1.75 0 0 0 0 0</pose>
    </include>

    <include>
      <name>three_disposal_bin</name>
      <uri>model://disposal_bin</uri>
      <pose>0 -0.9 0 0 0 0</pose>
    </include>

    <model name='camera'>
      <static>1</static>
      <pose>0.75 0 2.35 0 +1.5708 0</pose>
      <link name='camera_link'>
        <inertial>
          <mass>0.1</mass>
          <inertia>
            <ixx>0.000166667</ixx>
            <iyy>0.000166667</iyy>
            <izz>0.000166667</izz>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyz>0</iyz>
          </inertia>
        </inertial>
        <collision name='collision'>
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
          <max_contacts>10</max_contacts>
          <surface>
            <contact>
              <ode/>
            </contact>
            <bounce/>
            <friction>
              <torsional>
                <ode/>
              </torsional>
              <ode/>
            </friction>
          </surface>
        </collision>
        <visual name='visual'>
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <sensor name='camera' type='camera'>
          <camera>
            <horizontal_fov>1.047</horizontal_fov>
            <image>
              <width>800</width>
              <height>800</height>
            </image>
            <clip>
              <near>0.1</near>
              <far>100</far>
            </clip>
          </camera>
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <visualize>1</visualize>
          <plugin name='camera_controller' filename='libgazebo_ros_camera.so'>
            <alwaysOn>1</alwaysOn>
            <updateRate>10</updateRate>
            <cameraName>camera</cameraName>
            <imageTopicName>image_raw</imageTopicName>
            <cameraInfoTopicName>camera_info</cameraInfoTopicName>
            <frameName>camera_link</frameName>
            <hackBaseline>0.0</hackBaseline>
            <distortionK1>0.0</distortionK1>
            <distortionK2>0.0</distortionK2>
            <distortionK3>0.0</distortionK3>
            <distortionT1>0.0</distortionT1>
            <distortionT2>0.0</distortionT2>
            <CxPrime>0</CxPrime>
            <Cx>0.0</Cx>
            <Cy>0.0</Cy>
            <focalLength>0.0</focalLength>
          </plugin>
        </sensor>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
    </model>

  </world>
</sdf>
